abci/types/types.proto

```diff
@@ -238,10 +238,10 @@ message Header {
   // basic block info
   Version version = 1 [(gogoproto.nullable)=false];
   string chain_id = 2 [(gogoproto.customname)="ChainID"];
-  int64 height = 3;
```

Comment:
I hope its not too late to revert this decision prelaunch, but I think height being signed is the wrong decision. Feel free to close this comment if the ship has sailed.

Reply:
Did you read https://blog.cosmos.network/choosing-a-type-for-blockchain-height-beware-of-unsigned-integers-714804dddf1d ? Can you argue against it ? Having to add an `s` here made me doubt our decision for the first time, but I still think it's probably right.

Reply:
I have read it, will respond more later. I do disagree with it, and think that not choosing uint does increase chance of error. (I'll have to double check later, but I think we even have evidence of it doing so in our project)